src/domain/addContract.ts

```diff
@@ -37,7 +41,20 @@ export async function addContract(
   context: ChainContext,
   event: ConditionalOrderCreatedEvent
 ) {
-  return _addContract(context, event).catch(handleExecutionError);
+  const { chainId } = context;
+  const timer = addContractsRunDurationSeconds
+    .labels(chainId.toString())
+    .startTimer();
+  try {
+    await _addContract(context, event);
+  } catch (err) {
+    addContractsErrorsTotal
+      .labels(context.chainId.toString(), "addContract")
+      .inc();
+    handleExecutionError(err);
+  } finally {
+    timer();
```

Comment:
this pattern can be a good primitive, so maybe we can prevent repetition.

Do you think we can refactor it in sth like:

```ts
await measureTime({
  action: _addContract(context, event),
  timeMetric: addContractsRunDurationSeconds.labels(chainId.toString()),
  errorMetric: addContractsErrorsTotal
}).catch(handleExecutionError)
```

Reply:
Done